integrations/zoom/checks.go

```diff
@@ -46,13 +49,96 @@ func test(v sdkservices.Vars) sdkintegrations.OptFn {
 		case "":
 			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
 		case integrations.OAuthDefault, integrations.OAuthPrivate:
-			// TODO(INT-247): Implement.
+			return oauthTest(ctx, vs)
 		case integrations.ServerToServer:
-			// TODO(INT-247): Implement.
+			return ServerToServerTest(ctx, vs)
 		default:
 			return sdktypes.NewStatus(sdktypes.StatusCodeError, "Bad auth type"), nil
 		}
-
-		return sdktypes.NewStatus(sdktypes.StatusCodeOK, ""), nil
 	})
 }
+
+// oauthTest verifies the OAuth authentication with Zoom API using the "me" context.
+// (Based on: https://developers.zoom.us/docs/integrations/oauth/#the-me-context).
+func oauthTest(ctx context.Context, vs sdktypes.Vars) (sdktypes.Status, error) {
+	token := vs.GetValue(common.OAuthAccessTokenVar)
+	refreshToken := vs.GetValue(common.OAuthRefreshTokenVar)
+	clientID := os.Getenv("ZOOM_CLIENT_ID")
+	clientSecret := os.Getenv("ZOOM_CLIENT_SECRET")
+	url := "https://api.zoom.us/v2/users/me"
+
+	// First attempt with current token
+	resp, err := makeAPICall(ctx, token, url)
+	if err != nil {
+		return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		return sdktypes.NewStatus(sdktypes.StatusCodeOK, "OAuth connection successful"), nil
+	}
+
+	// Check if token expired
+	if resp.StatusCode == http.StatusUnauthorized && refreshToken != "" {
+		newToken, err := refreshTokenReq(ctx, refreshToken, clientID, clientSecret, vs)
+		if err != nil {
+			return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+		}
+
+		newResp, err := makeAPICall(ctx, newToken, url)
+		if err != nil {
+			return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+		}
+		defer newResp.Body.Close()
+
+		if newResp.StatusCode == http.StatusOK {
+			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "OAuth connection successful after token refresh"), nil
+		}
+
+		return sdktypes.NewStatus(sdktypes.StatusCodeError, "Failed to connect to Zoom API after token refresh"), nil
+	}
+
+	return sdktypes.NewStatus(sdktypes.StatusCodeError, "Failed to connect to Zoom API"), nil
+}
+
+// ServerToServerTest handles the Server-to-Server authentication test for Zoom
+// (Based on: https://developers.zoom.us/docs/internal-apps/s2s-oauth/).
+func ServerToServerTest(ctx context.Context, vs sdktypes.Vars) (sdktypes.Status, error) {
+	id := vs.GetValue(accountID)
+	clientID := vs.GetValue(clientID)
+	secret := vs.GetValue(clientSecret)
+
+	app := privateApp{
+		AccountID:    id,
+		ClientID:     clientID,
+		ClientSecret: secret,
+	}
+
+	// Get a server-to-server token
+	token, err := serverToken(ctx, &app)
+	if err != nil {
+		return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+	}
+
+	url := "https://api.zoom.us/v2/users"
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```

Comment:
Why not use `makeAPICall` (or `httpGet` according to my comment about this function name)